### PR TITLE
[2718] Delete paas prod users at the end of day

### PR DESCRIPTION
**Pull request by saliceti**
## What
Add option to delete the extra paas users

## How to review
Follow Readme

### CODE REVIEW

**Comment by saliceti** [comment]
[Delete paas prod users at the end of day](https://trello.com/c/YzUwaOJD/2718-delete-paas-prod-users-at-the-end-of-day)

File: scripts/check-users-in-space-developer-role.ps1 [hunk 1]
```diff
@@ -1,37 +1,81 @@
...
 
 $headers = @{
   Authorization = "$bearer"
   Accept        = "application/json"
 }
 
-$get_space_details = Invoke-RestMethod "https://api.london.cloud.service.gov.uk/v3/spaces?names=$space"  -H $headers
-$space_guid = $get_space_details.resources[0].guid
+$DELETE = $delete_arg -eq "delete"
+if ($DELETE) {
```
Comment (vigneshmsft) [inline scripts/check-users-in-space-developer-role.ps1:14]: remove is a better term than delete in this case?

File: README.md [hunk 2]
```diff
@@ -5,3 +5,17 @@ Shared code and documentation for Becoming a teacher infrastructure.
...
+
+List extra users:
+```
+% ./scripts/check-users-in-space-developer-role.ps1 <space_name> <slack_webhook_url>
+```
+
+Delete extra users:
+```
+% ./scripts/check-users-in-space-developer-role.ps1 <space_name> <slack_webhook_url> delete <protected_user_names_comma_separated_list>
+```
```
Comment (felixtheflex) [inline README.md:21]: i think it is worth adding that to get the list of parameters, you can do:-

`check-users-in-space-developer-role.ps1 - ` + tab will give the list of all input parameters too.

File: scripts/check-users-in-space-developer-role.ps1 [hunk 3]
```diff
@@ -1,37 +1,81 @@
 #!/usr/bin/env pwsh
 
-param([String] $space, [String] $slackwebhook)
-
-$data = @()
+param($space_name, $slackwebhook, $delete_arg, $protected_usernames_csv)
```
Comment (felixtheflex) [inline scripts/check-users-in-space-developer-role.ps1:3]: param([Parameter(Mandatory)] $space_name, $slackwebhook, $delete_arg, $protected_usernames_csv) ........?

File: scripts/check-users-in-space-developer-role.ps1 (hunk 1 above)
Comment (felixtheflex) [reply scripts/check-users-in-space-developer-role.ps1:14]: why not `if ($delete_arg -eq "delete")`............? and remove line 13?

File: scripts/check-users-in-space-developer-role.ps1 (hunk 3 above)
Comment (felixtheflex) [reply scripts/check-users-in-space-developer-role.ps1:3]: param($space_name, $slackwebhook, $delete_arg, $protected_usernames_csv='twd-bat-paas-prod@digital.education.gov.uk')

File: scripts/check-users-in-space-developer-role.ps1 (hunk 1 above)
Comment (saliceti) [reply scripts/check-users-in-space-developer-role.ps1]: The variable is used again for the logic later on

Comment (saliceti) [reply scripts/check-users-in-space-developer-role.ps1]: @vigneshmsft I've changed this parameter to `unset` and forced using named parameters for clarity

File: scripts/check-users-in-space-developer-role.ps1 (hunk 3 above)
Comment (saliceti) [reply scripts/check-users-in-space-developer-role.ps1]: Added [Parameter(Mandatory)] thanks

We agreed to not set bat default as it's a generic script

File: README.md (hunk 2 above)
Comment (saliceti) [reply README.md:21]: Tested and this only works in a powershell shell. The feature is of limited use

File: .github/workflows/check-users-in-space-developer-role.yml [hunk 4]
```diff
@@ -23,6 +23,10 @@ jobs:
 
       - name: Run powershell script
         run: |
-          ./scripts/check-users-in-space-developer-role.ps1 "${{ env.CF_SPACE_NAME }}" "${{secrets.SLACK_WEBHOOK}}"
+          ./scripts/check-users-in-space-developer-role.ps1 \
+            -space "${{ env.CF_SPACE_NAME }}" \
+            -slack_webhook "${{secrets.SLACK_WEBHOOK}}" \
+            -unset true \
+            -whitelist twd-bat-paas-prod@digital.education.gov.uk
```
Comment (felixtheflex) [inline .github/workflows/check-users-in-space-developer-role.yml:30]: should we be exposing this addy? better as a secret..?

**Review by vigneshmsft — APPROVED** [review]
tested 🚀

**Pull request merged**